server/api/product.js

```diff
@@ -0,0 +1,63 @@
+const router = require('express').Router()
+const { Product, Review, Category} = require('../db/models')
+
+router.param('productId', function (req, res, next, productId) {
+  Product.findById(productId)
+    .then(product => {
+      req.product = product;
+      next()
+    })
+})
+
+router.get('/', function (req, res, next) {
+  Product.findAll({
+    include: [{
+      model: Category
+    }]
+  })
+    .then(products => res.send(products))
+    .catch(next)
+})
+
+
+router.get('/:productId', function (req, res, next) {
+  Product.findAll({
+    where: { productid: req.params.productId },
```

Comment:
You use req.params.productId here but also put product on req.product. So you can use req.product.id or take out the router.param section above and use req.params.productId